metrics-exporter-prometheus/src/builder.rs

```diff
@@ -239,6 +243,38 @@ impl PrometheusBuilder {
         Ok(self)
     }
 
+    /// Sets the default bucket duration for rolling summaries
+    ///
+    /// Buckets will be cleared after this interval expires
+    ///
+    /// ## Errors
+    ///
+    /// If `value` less than 1 error will be thrown
+    pub fn set_bucket_duration(mut self, value: Duration) -> Result<Self, BuildError> {
+        if value.is_zero() {
+            return Err(BuildError::ZeroBucketDuration);
+        }
+
+        self.bucket_duration = Some(value);
+        Ok(self)
+    }
+
+    /// Sets the default bucket count for rolling summaries
+    ///
+    /// Count number buckets are created to store summary information
+    ///
+    /// ## Errors
+    ///
+    /// If `value` less than 1 error will be thrown
+    pub fn set_bucket_count(mut self, count: u32) -> Result<Self, BuildError> {
+        if count == 0 {
+            return Err(BuildError::ZeroBucketCount);
+        }
+
+        self.bucket_count = Some(count);
+        Ok(self)
```

Comment:
In conjunction with my comment a little further down about taking `NonZeroU32` directly in `Distribution::new_summary`, we should also make a corresponding change here... essentially, change `self.bucket_count` to `Option<NonZeroU32>` and then this code can become something like:

```rust
let count = NonZeroU32::new(count).ok_or(BuildError::ZeroBucketCount)?;

self.bucket_count = Some(count);
Ok(self)
```

Reply:
I changes the function signature to just take NonZeroU32 as an input param which removes the need for BuildError::ZeroBucketCount. Please let me know if that works for you.